sdk/synapse/Azure.Analytics.Synapse.Spark/tests/samples/Sample2_ExecuteSparkStatement.cs

```diff
@@ -56,5 +64,117 @@ public void ExecuteSparkStatementSync()
 
             Response operation = client.CancelSparkSession(sessionCreated.Id);
         }
+
+        // https://github.com/Azure/azure-sdk-for-net/issues/17587
+        // This code is copied from SparkTestUtilities.cs and modified, as there is no current way to monitor/poll for spark completion
+        // It belongs in a helper class, likely a LRO
+        // It is being left here only temporarily.
+        private const string Error = "error";
+        private const string Dead = "dead";
+        private const string Success = "success";
+        private const string Killed = "killed";
+        private const string Idle = "idle";
+
+       public static List<string> SessionSubmissionFinalStates = new List<string>
+        {
+            Idle,
+            Error,
+            Dead,
+            Success,
+            Killed
+        };
+
+        public static SparkSession PollSparkSession(
+            SparkSessionClient client,
+            SparkSession session,
+            IList<string> livyReadyStates = null)
+        {
+            if (livyReadyStates == null)
+            {
+                livyReadyStates = SessionSubmissionFinalStates;
+            }
+
+            return Poll(
+                session,
+                s => s.Result.ToString(),
+                s => s.State,
+                s => client.GetSparkSession(s.Id, true),
+                livyReadyStates);
+        }
+
+        private const string Starting = "starting";
+        private const string Waiting = "waiting";
+        private const string Running = "running";
+        private const string Cancelling = "cancelling";
+
+        private static List<string> ExecutingStates = new List<string>
+        {
+            Starting,
+            Waiting,
+            Running,
+            Cancelling
+        };
+
+        private static SparkStatement PollSparkStatement(
+            SparkSessionClient client,
+            int sessionId,
+            SparkStatement statement)
+        {
+            return Poll(
+                statement,
+                s => null,
+                s => s.State,
+                s => client.GetSparkStatement(sessionId, s.Id),
+                ExecutingStates,
+                isFinalState: false);
+        }
+
+        private static T Poll<T>(
+            T job,
+            Func<T, string> getJobState,
+            Func<T, string> getLivyState,
+            Func<T, T> refresh,
+            IList<string> livyReadyStates,
+            bool isFinalState = true,
+            int pollingInMilliseconds = 0,
+            int timeoutInMilliseconds = 0,
+            Action<T> writeLog = null)
+        {
+            var timeWaitedInMilliSeconds = 0;
```

Comment:
Could we use a CancellationToken for this?  I know you didn't write the original code, but it might end up being cleaner for the purposes of a sample.